### PR TITLE
Add README for PF-1.2 Policy-based traffic GRE Encapsulation to IPv4 GRE tunnel

### PR DESCRIPTION
**Pull request by sugrimov**

### CODE REVIEW

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces.
+
+2.  IPv4 and IPv6 static routes to test destinations networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  Static routes to 32x IPv4 GRE encap destinations towards ATE Port 2.
+
+4.  3 IPv4 and 3 IPv6 source prefixes will be used to generate tests traffic
```
Comment (daveruturaj) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:32]: we are using src1 and src2 only, then why do we need 3 sets of  source prefixes?

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces.
+
+2.  IPv4 and IPv6 static routes to test destinations networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  Static routes to 32x IPv4 GRE encap destinations towards ATE Port 2.
```
Comment (daveruturaj) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:30]: Do you mean another sets of IPv4 and IPv6 static routes  than the one in step 2 pointing to 32x IPv4 GRE tunnel destinations? If yes, then can you please specify?

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+*   ATE Port 2 is used as the destination port for encapsulated 
+    traffic.
+*   ATE Port 3 is used as the fallback destination for
+    pass-through traffic.
+
+### Configuration
+
+1.  All DUT Ports are configured as a singleton IP interfaces.
+
+2.  IPv4 and IPv6 static routes to test destinations networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
```
Comment (daveruturaj) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:28]: destination networks instead of destinations networks?

**Review by daveruturaj — APPROVED** [review]

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+2.  IPv4 and IPv6 static routes to test destination networks IPV4-DST/IPV6-DST are configured on DUT towards ATE Port 3.
+
+3.  Another set of IPv4 static routes to 32x IPv4 GRE encap destinations towards ATE Port 2.
+
+4.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+    - Match IPV6-SRC2 and encapsulate to 32 IPv4 GRE destinations.
```
Comment (dplore) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:37]: Please provide the example OC paths for these rules.   The paths mentioned at the end are missing things like the match rules.

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv4/config/source-address`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destinations`
+
+
+## Telemetry Parameter Coverage
+
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/state/matched-pkts`
+*   `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/state/matched-octets`
```
Comment (dplore) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:88]: Probably you need the match rule?  Also there are some other gre encap related paths, do you want to cover any of these (ok if you don't but since the basic match rules where missing, I thought I would list these for you to consider):

For example the match rule is: 

[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-config-identifying-prefix)

Here's a dump from https://openconfig.net/projects/models/paths/ matching the search string: "policy forwarding gre config/"




openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/config/decapsulate-gre | leaf | boolean
-- | -- | -- | --
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix | leaf | ip-prefix
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id | leaf | string
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source | leaf | ip-address
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination | leaf | ip-prefix
openconfig-network-instance | /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl | leaf | uint8


openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/config/decapsulate-gre](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-config-decapsulate-gre)	leaf	boolean
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-config-identifying-prefix)	leaf	ip-prefix
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-id)	leaf	string
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-source)	leaf	ip-address
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-destination)	leaf	ip-prefix
openconfig-network-instance	[/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl](https://openconfig.net/projects/models/schemadocs/yangdoc/openconfig-network-instance.html#network-instances-network-instance-policy-forwarding-policies-policy-rules-rule-action-encapsulate-gre-targets-target-config-ip-ttl)	leaf	uint8

**Review by dplore — COMMENTED** [review]
Please fix the failing CI check for README OpenConfig Path and RPC Coverage / integration-test.   

Also the test is missing the minimum required DUT platform?  (vrx, fff, mff).  See the template at: https://github.com/openconfig/featureprofiles/blob/main/doc/test-requirements-template.md#required-dut-platform

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+4.  2 IPv4 and 2 IPv6 source prefixes will be used to generate tests traffic 
+(SRC1-SRC2). Apply policy-forwarding with 4 rules to DUT Port 1:
+    - Match IPV4-SRC1 and accept/foward.
+    - Match IPV6-SRC1 and accept/foward.
+    - Match IPV4-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+    - Match IPV6-SRC2 and encapsulate to 32 IPv4 GRE destinations.
+
+5.   Set GRE encap source to device's loopback interface.
+6.   Either `identifying-prefix` or `targets/target/config/destination` can be used to configure GRE destinations.
```
Comment (dplore) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:45]: How should the test automation code decide which one to use?  Is this a deviation based on the NOS of the DUT?  (junos vs others?). I'm kind of surprised to see this in the OC model, but to get through this test given the current model, the README should specify a method to chose which path to use.

File: feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+    # match condition
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv4/config/source-address:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/ipv6/config/source-address:
+    # encap action
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/id:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/source:
+    # either destination or identifying-prefix can be specified
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/destination:
+    /network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/config/identifying-prefix:
+    # application to the interface
```
Comment (dplore) [inline feature/policy_forwarding/encapsulation/otg_tests/encap_gre_ipv4/README.md:126]: Do you want to include any coverage for `/network-instances/network-instance/policy-forwarding/policies/policy/rules/rule/action/encapsulate-gre/targets/target/config/ip-ttl`. (not strictly required if we aren't using this at the moment.

**Review by dplore — APPROVED** [review]
LGTM, left a couple of comments that are optional to address IMO.

**Review by dplore — APPROVED** [review]
